src/providers/consensus/client.py

```diff
@@ -57,22 +58,29 @@ def get_block_header(self, state_id: Union[str, SlotNumber, BlockRoot]) -> Block
     @lru_cache(maxsize=1)
     def get_block_details(self, state_id: Union[SlotNumber, BlockRoot]) -> BlockDetailsResponse:
         """Spec: https://ethereum.github.io/beacon-APIs/#/Beacon/getBlockV2"""
-        if state_id in self.NON_CACHEABLE_STATES:
-            raise ValueError(f'Block details for state_id: {state_id} could not be cached. '
-                             'Please provide slot number or block root.')
         data, _ = self._get(self.API_GET_BLOCK_DETAILS.format(state_id))
         return BlockDetailsResponse(**data)
 
     @lru_cache(maxsize=1)
-    def get_validators(self, *args, **kwargs) -> list[Validator]:
+    def get_validators(self, blockstamp: Union[BlockStamp], pub_keys: Optional[str | tuple] = None) -> list[Validator]:
         """Spec: https://ethereum.github.io/beacon-APIs/#/Beacon/getStateValidators"""
-        return self.get_validators_no_cache(*args, **kwargs)
+        return self.get_validators_no_cache(blockstamp, pub_keys)
 
     @list_of_dataclasses(Validator)
-    def get_validators_no_cache(self, state_id: Union[str, SlotNumber, StateRoot], pub_keys: Optional[str | tuple] = None) -> list[Validator]:
+    def get_validators_no_cache(
```

Comment:
Lets leave BlockStamp everywhere

Reply:
We use LiteralState only to get block_root

Reply:
Let's use blockstamp in all methods, except get_block_root method

Reply:
left `Union[SlotNumber, BlockRoot]` for `header` methods because we use them in `get_first_no_missed`